pipeline/api/cloud.py

```diff
@@ -89,6 +93,19 @@ def raise_for_invalid_token(self):
                 ),
             )
 
+    @staticmethod
+    def _get_raise_for_status(response: requests.Response) -> None:
+        # A handler for errors that might be sent with messages from Top.
+        if not response.ok:
+            content = response.json()
+            if "detail" in content.keys() and isinstance(content["detail"], Dict):
+                detail = content["detail"]
+                message = detail.pop("message", "Unidentified Error")
+                detail = detail or ""
+                raise Exception(f"Error {response.status_code}: {message} {detail}")
```

Comment:
The general behaviour here is nice, just a few comments on the details.

1. The idiomatic way of checking membership would be `"detail" in content`.
2. There are a lot of checks on the `detail` key.  I guess this is partly because Top does not have a consistent error message format? So sometimes it has `dict(detail="error message")` and sometimes it has `dict(detail=dict(message="error message"))`.

Reply:
Yes, as suspected some exceptions only send a string in detail. I will implement this. Also some exceptions only send a message with no other info

Reply:
PermissionNotGranted for example

Reply:
OK, that's a bit annoying, and no doubt an oversight on our part not to design a consistent error interface.

Something I forgot: it would be good to raise a more specific error here. ATM the code is so generic it'd be annoying to specialise the exception class per error, but perhaps having something like `class PipelineAPIRequestError(Exception): pass` would be a start?

Reply:
I'm just finding out how gross the exceptions are in Top